### ISSUE TITLE
Using state as a dependency for the onFileAction causes infinite loop

### ISSUE THREAD

**Issue by ttamminen** [TimboKZ/Chonky#52]
I need to update the component local state, for example, when a file selected. I get an infinite loop when I update the state in the `onFileAction` handler. Any ideas on how to implement the event handler, so that I could update the local state?

Here is an example that gives an infinite loop.
https://codesandbox.io/s/loving-clarke-j597s?file=/src/App.tsx

**Comment by TimboKZ** [comment]
Skimming through the code, I think these 2 changes should fix the issue:
1. Memoizing `files` and `folderChain` arrays using the `useMemo` hook.
2. Not using `myRandomState` as a dependency of `handleFileAction`. If you need to access the value of `myRandomState` in the handler, you can either use (a) functional `setState` or (b) "instance variable" made via `useRef` (also recall that Chonky provides the current selection to every action dispatch as `data.state.selectedFiles`).

**Comment by ttamminen** [comment]
I added the `useMemo` to the `files` and `folderChain`. 

Regarding point two, it is interesting that if I remove the `setMyRandomState` and/or `myRandomState` from the dependency array it still wants to call the callback. I updated the example to reflect this change. 

I think I'll try next to use the instance variable solution. I am a bit worried that there is something wrong with the `onFileAction` definition as referencing the state is probably a quite common operation. 

I was able to remove some of the selection related state references by using what is coming from the action dispatch. Thanks for the tip! 

My problem is that I need the selections outside of the Chonky context as I have a modal dialog's for example in the delete operation. When the user actives delete operation the Chonky action only actives the modal confirmation to be displayed and on it's `onConfirm` the delete is actually done.

```
<Modal
  isOpen={deleteModalOpen}
  onConfirm={async () => {
    // from the state
    selection.forEach((item) => {
      dispatch(deleteItem({ siteId, path: item }));
    });
    setDeleteModalOpen(false);
    await updateFileList({
      siteId,
      targetSiteVersion: siteVersion + selection.length,
      dispatch,
    });
  }}
  confirmButtonMessage={messages.delete}
  setOpen={setDeleteModalOpen}
  content={<DeleteConfirmation selections={selection} />}
/>;

```
